Source/SocketIO/Engine/SocketEngine.swift

```diff
@@ -250,6 +250,20 @@ open class SocketEngine : NSObject, URLSessionDelegate, SocketEnginePollable, So
         if client == nil {
             return (URL(string: "http://localhost/")!, URL(string: "http://localhost/")!)
         }
+        
+        if let scheme = URLComponents(string: url.absoluteString)!.scheme {
```

Comment:
I feel like there should be a way of doing this using the `url`, or by using `?` instead of `!`.

Reply:
Thank you for your patient. I will fix it soon.